File: src/main/java/etlmail/front/cli/NewsletterNotifier.java
package etlmail.front.cli;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;
import org.springframework.stereotype.Component;

import etlmail.engine.NewsletterNotification;

@Component
public class NewsletterNotifier {
	private static final Logger log = LoggerFactory
			.getLogger(NewsletterNotifier.class);

	private @Autowired
	NewsletterNotification newsletterNotification;

	public static void main(String[] args) {
		try {
			log.info("Initializing mail tool...");
			final AnnotationConfigApplicationContext ctx = new AnnotationConfigApplicationContext(
					CliAppCtx.class);
			ctx.registerShutdownHook();
			ctx.getBean(NewsletterNotifier.class).run();
		} catch (final Exception e) {
			log.error("Error, could not send mail", e);
			exitOnError();
		}
	}

	public void run() {
		log.info("Sending mail...");
		newsletterNotification.processNotification();
		log.info("Mail successfully sent");
		exitOnSuccess();
	}

	private static void exitOnError() {
		System.exit(1);
	}

	private static void exitOnSuccess() {
		System.exit(0);
	}

}

File: src/main/scala/etlmail/front/cli/NewsletterNotifier.scala
package etlmail.front.cli;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;
import org.springframework.stereotype.Component;

import com.weiglewilczek.slf4s.Logging

import etlmail.engine.NewsletterNotification;
import etlmail.engine.ToolMailSender;

@Component
class NewsletterNotifier extends Logging {
  import NewsletterNotifier._

  @Autowired private var toolMailSender: ToolMailSender = _

  @Autowired private var newsletterNotification: NewsletterNotification = _

  def run() {
    logger.info("Sending mail...")
    toolMailSender.sendMail(newsletterNotification)
    logger.info("Mail successfully sent")
    exitOnSuccess()
  }
}

object NewsletterNotifier extends App with Logging {
  try {
    logger.info("Initializing mail tool...")
    val ctx = new AnnotationConfigApplicationContext(classOf[CliAppCtx])
    ctx.registerShutdownHook()
    ctx.getBean(classOf[NewsletterNotifier]).run()
  } catch {
    case e: Exception =>
      logger.error("Error, could not send mail", e)
      exitOnError();
  }

  def exitOnError() {
    System.exit(1)
  }

  def exitOnSuccess() {
    System.exit(0)
  }
}
